scripts/kernel_bump.sh

```diff
@@ -0,0 +1,225 @@
+	else
+		_target_dir="target/linux/${platform_name}"
+	fi
+
+	if [ ! -d "${_target_dir}/image" ]; then
+		e_err 'Cannot find target linux directory.'
+		exit 1
+	fi
+
+	git switch --force-create '__openwrt_kernel_files_mover'
+
+	for _path in "${_target_dir}/"*; do
+		if [ ! -s "${_path}" ] || \
+		   [ "${_path}" = "${_path%%"-${source_version}"}" ]; then
+			continue
+		fi
+
+		_target_path="${_path%%"-${source_version}"}-${target_version}"
+		if [ -s "${_target_path}" ]; then
+			e_err "Target '${_target_path}' already exists!"
+			exit 1
+		fi
+
+		git mv \
+			"${_path}" \
+			"${_target_path}"
+	done
+
+	find "${_target_dir}" -iname "config-${source_version}" | while read -r _config; do
+		_path="${_config%%"/config-${source_version}"}"
+		git mv "${_config}" "${_path}/config-${target_version}"
+	done
+
+	git commit \
+		--signoff \
+		--message "kernel/${platform_name}: Create kernel files for v${target_version} (from v${source_version})" \
+		--message 'This is an automatically generated commit.' \
+		--message 'During a `git bisect` session, `git bisect --skip` is recommended.'
+
+	git checkout 'HEAD~' "${_target_dir}"
+	git commit \
+		--signoff \
+		--message "kernel/${platform_name}: Restore kernel files for v${source_version}" \
+		--message "$(printf "This is an automatically generated commit which aids following Kernel patch history,\nas git will see the move and copy as a rename thus defeating the purpose.\n\nSee: https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html\nfor the original discussion.")"
+	git switch "${initial_branch:?Unable to switch back to original branch. Quitting.}"
+	GIT_EDITOR=true git merge --no-ff '__openwrt_kernel_files_mover'
+	git branch --delete '__openwrt_kernel_files_mover'
+
+	echo "Original commitish was '${initial_commitish}'."
+	echo 'Kernel bump complete. Remember to use `git log --follow`.'
+}
+
+check_requirements()
+{
+	for _cmd in ${REQUIRED_COMMANDS}; do
+		if ! _test_result="$(command -V "${_cmd}")"; then
+			_test_result_fail="${_test_result_fail:-}${_test_result}\n"
+		else
+			_test_result_pass="${_test_result_pass:-}${_test_result}\n"
+		fi
+	done
+
+	echo 'Available commands:'
+	# As the results contain \n, we expect these to be interpreted.
+	# shellcheck disable=SC2059
+	printf "${_test_result_pass:-none\n}"
+	echo
+	echo 'Missing commands:'
+	# shellcheck disable=SC2059
+	printf "${_test_result_fail:-none\n}"
+	echo
+
+	if [ -n "${_test_result_fail:-}" ]; then
+		echo 'Command test failed, missing programs.'
+		test_failed=1
+	fi
+}
+
+main()
+{
+	while getopts 'hp:s:t:' _options; do
+		case "${_options}" in
+		'h')
+			usage
+			exit 0
+			;;
+		'p')
+			platform_name="${OPTARG}"
+			;;
+		's')
+			source_version="${OPTARG#v}"
+			;;
+		't')
+			target_version="${OPTARG#v}"
+			;;
+		':')
+			e_err "Option -${OPTARG} requires an argument."
+			exit 1
+			;;
+		*)
+			e_err "Invalid option: -${OPTARG}"
+			exit 1
+			;;
+		esac
+	done
+	shift "$((OPTIND - 1))"
```

Comment:
Since all 3 are **required**, why bother having them as options?  Easier for use to simply have them as 3 arguments with required order.

Reply:
Because it's always better to be explicit rather then implicit. It's nice that we are not bound to a fixed argument order. It's nice we make it specific what argument is for what purpose. It's nice if we want to add options, we don't have to worry about order and location.